File: godfather/roles/dethy/naive_cop.py
from godfather.roles.town.cop import Cop


class NaiveCop(Cop):
    """
    The law enforcer of town, believes he’s the best cop and nothing bad would ever happen to the town.

    - Win Condition: Lynch every criminal and evildoer

    + Abilities: Interrogate one person each night for suspicious activity
    + You will know if your target is suspicious or innocent

    Special Interactions: A naive cop will always show innocent results regardless of the roles]
    """
    name = 'Naive Cop'

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.categories.remove('Town Investigative')

    def result_modifier(self, _innocence):
        return True

File: src/lib/mafia/roles/town/dethy/Naive_Cop.ts
import Cop from '../Cop';

// @ts-ignore weird error
class Naive_Cop extends Cop {
  public name = 'Naive Cop';

  public innocenceModifier() {
    return true;
  }
}

Naive_Cop.categories = ['Dethy Cop'];

export default Naive_Cop;
